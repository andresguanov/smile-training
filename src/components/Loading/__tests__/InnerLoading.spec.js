import { shallowMount } from "@vue/test-utils"
import smInnerLoading from "@components/Loading/InnerLoading.vue"
let wrapper

describe("smInnerLoading", () => {
  //#region Basic render
  it("renders default loading", () => {
    wrapper = shallowMount(smInnerLoading, {
      propsData: {
        value: true
      }
    })
    expect(wrapper.classes("loading")).toBe(true)
    expect(wrapper.classes("absolute")).toBe(false)
    expect(wrapper.find(".msg").text()).toBe(wrapper.vm.$options.props.text.default)
    expect(wrapper.html()).toMatchSnapshot()
  })
  //#endregion

  //#region Props
  it("sets absolute", () => {
    wrapper = shallowMount(smInnerLoading, {
      propsData: {
        value: true,
        absolute: true
      }
    })
    expect(wrapper.classes("absolute")).toBe(true)
  })

  it("sets text", () => {
    wrapper = shallowMount(smInnerLoading, {
      propsData: {
        value: true,
        text: "Loading"
      }
    })
    expect(wrapper.find(".msg").text()).toBe("Loading")
  })
  //#endregion
})
