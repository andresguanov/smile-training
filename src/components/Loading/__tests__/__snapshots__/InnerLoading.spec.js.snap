// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smInnerLoading renders default loading 1`] = `
<div class="loading">
  <div class="msg-inner">
    <sm-progress-circle-stub size="small" value="0" max="100"></sm-progress-circle-stub>
    <div class="msg">
      Buscando...
    </div>
  </div>
</div>
`;
